src/builtin_argparse.cpp

```diff
@@ -535,9 +535,9 @@ static int handle_flag(parser_t &parser, const argparse_cmd_opts_t &opts, option
         // short or long flag was given.
         assert(!woptarg);
         if (long_idx == -1) {
-            opt_spec->vals.push_back(wcstring(1, L'-') + opt_spec->short_flag);
+            opt_spec->vals.emplace_back(1, L'-' + opt_spec->short_flag);
```

Comment:
I don't think this is correct - the original was just a cheesy way of saying "Make a string containing `-`", via the "fill" constructor. I'm not sure which constructor this is now supposed to use, but it appears to not be working - see the failing tests.

Reply:
Right. It's not working because of thr 1. I'll fix.